support-services/src/main/scala/com/gu/support/pricing/PriceSummaryService.scala

```diff
@@ -12,20 +12,30 @@ import scala.math.BigDecimal.RoundingMode
 
 class PriceSummaryService(promotionService: PromotionService, catalogService: CatalogService) extends TouchpointService {
   private type GroupedPriceList = Map[(FulfilmentOptions, ProductOptions, BillingPeriod), Map[Currency, PriceSummary]]
+  case class PromotionWithCode(promoCode: PromoCode, promotion: Promotion)
+
+  private def findPromotion(maybePromoCode: Option[PromoCode]) =
+    for {
+      promoCode <- maybePromoCode
+      promotion <- promotionService.findPromotion(promoCode)
```

Comment:
this will still be scanning the whole table (although only once in this case), I was under the impression the View table was there to allow lookup by promo code in the checkout.  I can see that based on the amount of data we have it's not too bad, but in that case do we need the View table at all?
This is all a bit fuzzy from recollection so I may have missed something else.

Reply:
Unfortunately the view table doesn't have required fields in it for our purposes. We could look at adding them and switching to using the view table in a future PR if it seems useful.